network/transport/grpc/connection.go

```diff
@@ -107,6 +113,7 @@ type conn struct {
 	peer             atomic.Value
 	ctx              context.Context
 	cancelCtx        func()
+	errStatus        *status.Status
```

Comment:
you could use a `atomic.Pointer` (with generics), to avoid having to use the mutex;

```go
status atomic.Pointer[status.Status]
```